times_mapping.txt

```diff
@@ -2,7 +2,8 @@ ACT_BND[REG,DATAYEAR,PRC,TSLVL,BD,VALUE] = ~FI_T(Region,Year,TechName,TimeSlice,
 ACT_COST[REG,DATAYEAR,PRC,CUR,VALUE] = ~FI_T(Region,Year,TechName,Curr,Attribute:ACT_COST)
 ACT_COST[REG,DATAYEAR,PRC,CUR,VALUE] = ~FI_T(Region,Year,TechName,Curr,Attribute:ACTCOST)
 ACT_CUM[REG,PRC,DATAYEAR,DATAYEAR2,BD,VALUE] = ~FI_T(Region,TechName,Year,Year2,LimType,Attribute:CUM)
-ACT_EFF[REG,DATAYEAR,PRC,COM,TSLVL,VALUE] = ~FI_T(Region,Year,TechName,Comm-IN,TimeSlice,Attribute:CEFF)
+ACT_EFF[REG,DATAYEAR,PRC,COM,TSLVL,VALUE] = ~FI_T(Region,Year,TechName,CommName,TimeSlice,Attribute:CEFF)
+ACT_EFF[REG,DATAYEAR,PRC,COM,TSLVL,VALUE] = ~FI_T(Region,Year,TechName,Other_Indexes,TimeSlice,Attribute:EFF)
```

Comment:
We have data specified for CEFF and EFF in the files, but both of them are Veda aliases for ACT_EFF which involve use of different defaults. For CEFF one has to specify the name of a commodity group in other_indexes that includes only one commodity (and is the same as that commodity name), while for EFF one specifies `ACT`.
Would be great to come up with a data structure that could hold this info, so we could avoid rows like these in the mapping file.

Reply:
If I'm reading the code in `produce_times_tables` correctly, I think the if you have two mappings in this file with the same `times_name` (ACT_EFF, in this case) then the table generated by the second mapping overwrites the one from the first in `result`:
https://github.com/samwebster/times-excel-reader/blob/0e16517b6873b39e4a93a7ab3fc4eeb29db9e00c/times_reader/main.py#L307

So I don't think having two mappings right now does what we want. Perhaps an issue/PR to add the desired functionality? I agree though, it would be nicer to have a proper representation rather than an ad-hoc duplicate mapping.

Reply:
Interesting! Mostly because this is not the first duplicate mapping we are using. 🤔
I did try another variant of it where only a column name on the right side was different and that did not work.
In any case would be great to check! Anyway the current way is probably temporary, because we would need around 300 unique entries on the left side to include all the TIMES parameters. And then there are aliases...